src/sql/workbench/contrib/queryplan2/browser/queryPlan.ts

```diff
@@ -0,0 +1,594 @@
+	public readonly title = localize('queryPlanTitle', "Query Plan");
+	public readonly identifier = 'QueryPlan2Tab';
+	public readonly view: QueryPlan2View;
+
+	constructor() {
+		this.view = new QueryPlan2View();
+	}
+
+	public dispose() {
+		dispose(this.view);
+	}
+
+	public clear() {
+		this.view.clear();
+	}
+}
+
+export class QueryPlan2View implements IPanelView {
+	private _qps?: QueryPlan2[] = [];
+	private _graphs?: azdata.QueryPlanGraph[] = [];
+	private _container = DOM.$('.qp2-container');
+
+	public render(container: HTMLElement): void {
+		container.appendChild(this._container);
+		this._container.style.overflow = 'scroll';
+	}
+
+	dispose() {
+		this._container.remove();
+		delete this._qps;
+		delete this._graphs;
+	}
+
+	public layout(dimension: DOM.Dimension): void {
+		this._container.style.width = dimension.width + 'px';
+		this._container.style.height = dimension.height + 'px';
+	}
+
+	public clear() {
+		this._qps = [];
+		this._graphs = [];
+		DOM.clearNode(this._container);
+	}
+
+	public addGraphs(newGraphs: azdata.QueryPlanGraph[]) {
+		newGraphs.forEach(g => {
+			const qp2 = new QueryPlan2(this._container, this._qps.length + 1);
+			qp2.graph = g;
+			this._qps.push(qp2);
+			this._graphs.push(g);
+			this.updateRelativeCosts();
+		});
+	}
+
+	private updateRelativeCosts() {
+		const sum = this._graphs.reduce((prevCost: number, cg) => {
+			return prevCost += cg.root.subTreeCost + cg.root.cost;
+		}, 0);
+
+		this._qps.forEach(qp => {
+			qp.relativeCost = ((qp.graph.root.subTreeCost + qp.graph.root.cost) / sum) * 100;
+		});
+	}
+}
+
+export class QueryPlan2 {
+	private _graph?: azdata.QueryPlanGraph;
+	private _relativeCost?: globalThis.Text;
+	private _actionBar: ActionBar;
+	private _table: Slick.Grid<any>;
+	private _dataView: Slick.Data.DataView<any>;
+	private _container: HTMLElement;
+	private _actionBarContainer: HTMLElement;
+	private _data: any[];
+	private _iconMap: any = new Object();
+	private _iconPaths: any = new Object();
+
+	public propContainer: HTMLElement;
+
+	constructor(
+		parent: HTMLElement,
+		private _graphIndex: number,
+
+	) {
+		this._container = DOM.$('.query-plan2-container');
+		parent.appendChild(this._container);
+
+
+		this._actionBarContainer = DOM.$('.qp2-actionbar-container');
+		this._actionBar = new ActionBar(this._actionBarContainer, {
+			orientation: ActionsOrientation.VERTICAL, context: this
+		});
+
+		this.propContainer = DOM.$('.properties-container');
+		const propHeader = document.createElement('div');
+		propHeader.className = 'properties-header';
+		propHeader.innerText = 'Properties';
+		this.propContainer.appendChild(propHeader);
+
+		this.propContainer.style.visibility = 'hidden';
+
+		this._dataView = new Slick.Data.DataView({ inlineFilters: false });
+		let self = this;
+		this._data = [];
+		const TaskNameFormatter = function (row, cell, value, columnDef, dataContext) {
+			value = value.replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
```

Comment:
there is already a utility method for this purpose, use the escape() method in src\vs\base\common\strings.ts